File: Java/Analyse des éléments d’un tableau d’entiers (pairs, impairs et zéros).java
import java.util.Scanner;

public class AnalyseTableau {

    // Méthode qui calcule le nombre de pairs, impairs et zéros
    public static void fct3(int[] t, int n, int[] resultats) {
        int np = 0, nimp = 0, nbz = 0;

        for (int i = 0; i < n; i++) {
            if (t[i] == 0)
                nbz++;
            else if (t[i] % 2 == 0)
                np++;
            else
                nimp++;
        }

        // Stockage des résultats
        resultats[0] = nbz;
        resultats[1] = np;
        resultats[2] = nimp;
    }

    public static void main(String[] args) {

        Scanner clavier = new Scanner(System.in);

        System.out.print("Entrer la dimension du tableau : ");
        int N = clavier.nextInt();

        int[] T = new int[N];

        for (int i = 0; i < N; i++) {
            System.out.print("Entrer T[" + i + "] : ");
            T[i] = clavier.nextInt();
        }

        int[] resultats = new int[3]; // [0]=zéros, [1]=pairs, [2]=impairs

        fct3(T, N, resultats);

        System.out.println("\n--- Résultats ---");
        System.out.println("Nombre de zéros : " + resultats[0]);
        System.out.println("Nombre d’éléments pairs : " + resultats[1]);
        System.out.println("Nombre d’éléments impairs : " + resultats[2]);

        clavier.close();
    }
}

File: C/Analyse des éléments d’un tableau d’entiers (pairs, impairs et zéros).c
#include <stdio.h>

/* Fonction qui calcule :
   - np   : nombre d’éléments pairs
   - nimp : nombre d’éléments impairs
   - nbz  : nombre de zéros
*/
void fct3(int t[], int n, int *np, int *nimp, int *nbz) {
    int i;

    for (i = 0; i < n; i++) {
        if (t[i] == 0) {
            (*nbz)++;
        } else if (t[i] % 2 == 0) {
            (*np)++;
        } else {
            (*nimp)++;
        }
    }
}

int main() {
    int N, i;
    int NP = 0, NIMP = 0, NBZ = 0;
    int T[50];

    printf("Entrer la dimension de T : ");
    scanf("%d", &N);

    for (i = 0; i < N; i++) {
        printf("Entrer T[%d] : ", i);
        scanf("%d", &T[i]);
    }

    /* Appel de la fonction */
    fct3(T, N, &NP, &NIMP, &NBZ);

    printf("\nLe nombre de zéros : %d\n", NBZ);
    printf("Le nombre d’elements pairs : %d\n", NP);
    printf("Le nombre d’elements impairs : %d\n", NIMP);

    return 0;
}
